### PR TITLE
Fix colorbar orientation to always be min->max

### PR DESCRIPTION
**Pull request by djhoese**
A user of one of my projects noticed that the colors for vertical colorbars (scales) had the colors flipped from what they should be. As far as I can tell this has always been like this. Additionally, the colormap was always assumed to have increasing values, but in recent versions of trollimage it is possible to have values in reverse order (flipped values versus flipped colors). This produces colorbars that don't follow what I think most people would expect which is for the colorbar to go from the minimum value to the maximum value. In the horizontal orientation this means minimum value on the left, maximum on the right. For vertical orientation this means minimum value on the bottom, maximum on the top. This PR fixes all of this.

montage

### CODE REVIEW

**Comment by djhoese** [comment]
Oh I also added pre-commit.ci to this repository. We were already using pre-commit-config in the project, but I forgot to enable the repository on pre-commit.ci.

**Comment by djhoese** [comment]
Ok I've updated the original description with example images. These are using the "rdbu" colormap so the first color is red, the last color is blue.

File: pydecorate/decorator_base.py [hunk 1]
```diff
@@ -748,21 +755,26 @@ def _insert_RGBA_image(self, img, box):
...
             "Missing 'trollimage' dependency. Required colorbar creation."
         )
 
     if is_vertical:
+        # Image arrays have row 0 being the top of the image so we must flip all calculations
         linedata = np.ones((scale_width, 1)) * np.arange(
-            minval, maxval, float(maxval - minval) / scale_height
+            max_cmap_value,
+            min_cmap_value,
+            float(min_cmap_value - max_cmap_value) / scale_height,
```
Comment (mraspaud) [inline pydecorate/decorator_base.py:770]: `np.linspace(max_cmap_value, min_cmap_value, scale_height, endpoint=False)` is equivalent to the `arange` and also clearer imo

File: pydecorate/decorator_base.py [hunk 2]
```diff
@@ -748,21 +755,26 @@ def _insert_RGBA_image(self, img, box):
...
         )
         linedata = linedata.transpose()
     else:
         linedata = np.ones((scale_height, 1)) * np.arange(
-            minval, maxval, float(maxval - minval) / scale_width
+            min_cmap_value,
+            max_cmap_value,
+            float(max_cmap_value - min_cmap_value) / scale_width,
```
Comment (mraspaud) [inline pydecorate/decorator_base.py:777]: Can also be replaced with `np.linspace`

**Review by mraspaud — APPROVED** [review]
LGTM, Just a suggestion to use linspace instead of arange

File: pydecorate/decorator_base.py (hunk 1 above)
Comment (djhoese) [reply pydecorate/decorator_base.py:770]: I have a feeling when these were originally written the endpoint keyword argument didn't exist...or the person didn't know about linspace. Let me do some tests to make sure it produces the same then but yeah agreed it should overall be clearer.

**Comment by djhoese** [comment]
As mentioned on slack, here is what this particular colormap looks like with the new HCL logic in trollimage:

image


About the same in my opinion, but maybe a little smoother/cleaner on the red to white transition.

**Pull request merged**